src/wallet/wallet_tx_builder.h

```diff
@@ -17,12 +22,322 @@ class ResolvedPayment : public RecipientMapping {
 public:
     CAmount amount;
     std::optional<Memo> memo;
+    bool isInternal{false};
 
     ResolvedPayment(
             std::optional<libzcash::UnifiedAddress> ua,
             libzcash::RecipientAddress address,
             CAmount amount,
-            std::optional<Memo> memo) :
-        RecipientMapping(ua, address), amount(amount), memo(memo) {}
+            std::optional<Memo> memo,
+            bool isInternal) :
+        RecipientMapping(ua, address), amount(amount), memo(memo), isInternal(isInternal) {}
+};
+
+/**
+ * A requested payment that has not yet been resolved to a
+ * specific recipient address.
+ */
+class Payment {
+private:
+    PaymentAddress address;
+    CAmount amount;
+    std::optional<Memo> memo;
+    bool isInternal;
```

Comment:
How can the user request a payment to an internal address, if we never intend to expose internal addresses to users? This suggests that either `Payment` is not storing the thing it appears to claim, or we are not correctly enforcing that users cannot specify internal addresses as arguments to `z_sendmany`.

Reply:
Ok, so the change I made is to remove `isInternal` from `Payment`, but leave it in `ResolvedPayment`. Payment should be the user-provided address, and then `ResolvedPayment` is more a detail of the tx builder. `Payment` always defaulted `isInternal` to false, but `ResolvedPayment` has `isInternal = true` for change addresses.